api/src/main.rs

```diff
@@ -175,37 +175,31 @@ fn read_token_file(path: &PathBuf) -> Result<SecUtf8, StartUpError> {
     Ok(SecUtf8::from(token))
 }
 
-fn read_ca_file(path: &PathBuf) -> Result<Certificate, StartUpError> {
+fn read_ca_file(path: &PathBuf) -> Result<Vec<X509>, StartUpError> {
     debug!(
         "Reading certificate authority from {}.",
         path.to_str().unwrap()
     );
 
-    let mut f = match File::open(path) {
-        Ok(f) => f,
-        Err(e) => {
-            return Err(StartUpError::CannotReadCertificateAuthority {
-                path: String::from(path.to_str().unwrap()),
-                err: format!("{}", e),
-            });
-        }
-    };
+    let mut file =
+        File::open(path).map_err(|err| StartUpError::CannotReadCertificateAuthority {
+            path: String::from(path.to_str().unwrap()),
+            err: format!("{}", err),
+        })?;
 
-    let mut buf = Vec::new();
-    if let Err(e) = f.read_to_end(&mut buf) {
-        return Err(StartUpError::CannotReadCertificateAuthority {
+    let mut buffer = Vec::new();
+    file.read_to_end(&mut buffer)
```

Comment:
The `?` operator is missing here or am I blind?

Reply:
Please run `cargo clippy` to double check. This will print many warnings because I was not aware of it in the beginning of PREvant (I was a Rust beginner) and the code is not yet idiomatic Rust in all places. But I try to improve with every commit. So maybe you will get some hints for your code changes (you can ignore the rest and please, do not try to apply all hints because otherwise the PR will explode 😅 🙈 ).

Reply:
You're right about the `read_to_end()`! I remember the compiler warning me about the `#[must_use]`, but that disappeared when I added the `map_err` even though I forgot the `?` 😰

And … yeah, making Clippy happy would be a nice task for another PR. Seems like a great way to learn :D

Reply:
Okay, I think the `must_use` warning just went by me without me noticing it.

Should we add `#![deny(unused_must_use)]`?